### PR TITLE
Philips hue sensor lux support

### PR DESCRIPTION
**Pull request by JordyThien**
Requested by the community
Also fixed the temperature (no decimal points were shown, now it is)

### CODE REVIEW

File: hardware/PhilipsHue/PhilipsHue.cpp [hunk 1]
```diff
@@ -936,14 +943,32 @@ void CPhilipsHue::ReportTemperature(const int NodeID, const float temperature, c
 	tsen.TEMP.id1 = (sID & 0xFF00) >> 8;
```
Comment (gizmocuz) [inline hardware/PhilipsHue/PhilipsHue.cpp]: could you please remove this, and use the default way of reporting a (temperature) sensor ?
Just call

`void SendTempSensor(const int NodeID, const int BatteryLevel, const float temperature, const std::string &defaultname);
`

These functions are specified in DomoticzHardware.cpp/h

File: hardware/PhilipsHue/PhilipsHue.cpp [hunk 2]
```diff
@@ -936,14 +943,32 @@ void CPhilipsHue::ReportTemperature(const int NodeID, const float temperature, c
...
+	int at10 = round(abs(temperature*10.0f));
 	tsen.TEMP.temperatureh = (BYTE)(at10 / 256);
 	at10 -= (tsen.TEMP.temperatureh * 256);
 	tsen.TEMP.temperaturel = (BYTE)(at10);
 
 	sDecodeRXMessage(this, (const unsigned char *)&tsen.TEMP, Name.c_str(), BatteryLevel);
 }
 
+
+void CPhilipsHue::ReportLux(const int NodeID, const float lux, const string &Name, uint8_t BatteryLevel)
```
Comment (gizmocuz) [inline hardware/PhilipsHue/PhilipsHue.cpp]: could you please remove this, and use the default way of reporting a (lux) sensor ?
Just call

`void SendLuxSensor(const int NodeID, const int ChildID, const int BatteryLevel, const float Lux, const std::string &defaultname);
`

These functions are specified in DomoticzHardware.cpp/h

**Review by gizmocuz — COMMENTED** [review]
your fast! ;) two requests

File: hardware/PhilipsHue/PhilipsHue.cpp (hunk 1 above)
Comment (JordyThien) [reply hardware/PhilipsHue/PhilipsHue.cpp]: Good point! Didn't knew about those functions. Updated the code

**Comment by gizmocuz** [comment]
Great! Thanks!!

**Comment by roelbeun** [comment]
Curious, when will this commit be available in the beta or even the master? Thanks for your work!

**Comment by JordyThien** [comment]
@Ruvetuve it's already in the beta release. You can download the latest beta on http://domoticz.com/downloads/ or run ./updatebeta

**Comment by roelbeun** [comment]
Great, thanks! Hope it comes to the stable release soon, in the mean while ill test it on a separate system.

**Pull request merged**